spring-batch-infrastructure/src/main/java/org/springframework/batch/item/file/builder/FlatFileItemWriterBuilder.java

```diff
@@ -234,14 +249,245 @@
+		 *
+		 * @param names names of each field
+		 * @return The parent {@link FlatFileItemWriterBuilder}
+		 * @see BeanWrapperFieldExtractor#setNames(String[])
+		 */
+		public FlatFileItemWriterBuilder<T> names(String[] names) {
+			this.names.addAll(Arrays.asList(names));
+			return this.parent;
+		}
+
+		public FormatterLineAggregator<T> build() {
+			Assert.notNull(this.format, "A format is required");
+			Assert.isTrue((this.names != null && !this.names.isEmpty()) || this.fieldExtractor != null,
+					"A list of field names or a field extractor is required");
+
+			FormatterLineAggregator<T> formatterLineAggregator = new FormatterLineAggregator<>();
+			formatterLineAggregator.setFormat(this.format);
+			formatterLineAggregator.setLocale(this.locale);
+			formatterLineAggregator.setMinimumLength(this.minimumLength);
+			formatterLineAggregator.setMaximumLength(this.maximumLength);
+
+			if (this.fieldExtractor == null) {
+				BeanWrapperFieldExtractor<T> beanWrapperFieldExtractor = new BeanWrapperFieldExtractor<>();
+				beanWrapperFieldExtractor.setNames(this.names.toArray(new String[this.names.size()]));
+				try {
+					beanWrapperFieldExtractor.afterPropertiesSet();
+				}
+				catch (Exception e) {
+					throw new IllegalStateException("Unable to initialize FormatterLineAggregator", e);
+				}
+				this.fieldExtractor = beanWrapperFieldExtractor;
+			}
+
+			formatterLineAggregator.setFieldExtractor(this.fieldExtractor);
+			return formatterLineAggregator;
+		}
+	}
+
+	/**
+	 * A builder for constructing a {@link DelimitedLineAggregator}
+	 *
+	 * @param <T> the type of the parent {@link FlatFileItemWriterBuilder}
+	 */
+	public static class DelimitedBuilder<T> {
+
+		private FlatFileItemWriterBuilder<T> parent;
+
+		private List<String> names = new ArrayList<>();
+
+		private String delimiter = ",";
+
+		private FieldExtractor<T> fieldExtractor;
+
+		protected DelimitedBuilder(FlatFileItemWriterBuilder<T> parent) {
+			this.parent = parent;
+		}
+
+		/**
+		 * Define the delimiter for the file.
+		 *
+		 * @param delimiter String used as a delimiter between fields.
+		 * @return The instance of the builder for chaining.
+		 * @see DelimitedLineAggregator#setDelimiter(String)
+		 */
+		public DelimitedBuilder<T> delimiter(String delimiter) {
+			this.delimiter = delimiter;
+			return this;
+		}
+
+		/**
+		 * Names of each of the fields within the fields that are returned in the order
+		 * they occur within the delimited file. These names will be used to create
+		 * a {@link BeanWrapperFieldExtractor} only if no explicit field extractor
+		 * is set via {@link DelimitedBuilder#fieldExtractor(FieldExtractor)}.
+		 *
+		 * @param names names of each field
+		 * @return The parent {@link FlatFileItemWriterBuilder}
+		 * @see BeanWrapperFieldExtractor#setNames(String[])
+		 */
+		public FlatFileItemWriterBuilder<T> names(String[] names) {
+			this.names.addAll(Arrays.asList(names));
+			return this.parent;
+		}
```

Comment:
It would be nice to have a List option as well...

Reply:
I took the same approach as the `FlatFileItemReaderBuilder` to be consistent.

We can add this for both reader and writer builders in another step.